### PR TITLE
feat: Add initial skeleton for ModsView

### PR DESCRIPTION
**Pull request by GeckoEidechse**
Precedes #17

### CODE REVIEW

File: src-tauri/src/lib.rs [hunk 1]
```diff
@@ -471,3 +477,37 @@ pub fn set_mod_enabled_status(
...
+    // Check if valid JSON and parse
+    let res: serde_json::Value = match serde_json::from_str(&data) {
+        Ok(res) => res,
+        Err(err) => return Err(err.to_string()),
+    };
+
+    let mut installed_mods = Vec::new();
+
+    for (key, value) in res.as_object().unwrap() {
+        println!("{:?} ===> {:?}", key, value);
```
Comment (Alystrasz) [inline src-tauri/src/lib.rs:500]: You should remove this.

File: src-vue/src/views/DeveloperView.vue [hunk 2]
```diff
@@ -112,6 +116,33 @@ export default defineComponent({
...
                     });
                 });
+        },
+        async getInstalledMods() {
+            let game_install = {
+                game_path: this.$store.state.game_path,
+                install_type: this.$store.state.install_type
+            } as GameInstall;
+            await invoke("get_installed_mods_caller", { gameInstall: game_install }).then((message) => {
+                // Simply console logging for now
```
Comment (Alystrasz) [inline src-vue/src/views/DeveloperView.vue:126]: So this will get removed in next PR, right?

File: src-tauri/src/lib.rs [hunk 3]
```diff
@@ -471,3 +477,37 @@ pub fn set_mod_enabled_status(
 
     Ok(())
 }
+
+/// Gets list of installed mods and their properties
+/// - name
+/// - is enabled?
+pub fn get_installed_mods(game_install: GameInstall) -> Result<Vec<NorthstarMod>, String> {
+    let enabled_mods_json_path = format!("{}/R2Northstar/enabledmods.json", game_install.game_path);
```
Comment (Alystrasz) [inline src-tauri/src/lib.rs:485]: How is this working on Windows with wrong path separators?

**Review by Alystrasz — CHANGES_REQUESTED** [review]

File: src-tauri/src/lib.rs (hunk 3 above)
Comment (GeckoEidechse) [reply src-tauri/src/lib.rs:485]: Idk, it kinda just does ¯\\\_(ツ)\_/¯

And this is not the first time I've used forward slashes in the codebase:
https://github.com/GeckoEidechse/FlightCore/blob/5cc82fb80f5f91ed9d4a32a71e090a9a805ff2a9/src-tauri/src/lib.rs#L208


We could look into using the right path separator for the right OS but for this PR I think it's fine to keep using forward slashes. ^^

File: src-vue/src/views/DeveloperView.vue (hunk 2 above)
Comment (GeckoEidechse) [reply src-vue/src/views/DeveloperView.vue:126]: Not the next one directly but once the feature is stabilised and release, then yes.

File: src-tauri/src/lib.rs (hunk 1 above)
Comment (GeckoEidechse) [reply src-tauri/src/lib.rs]: Done with 6148dd0945786b93473b5f58201b0671bcd2ec96

File: src-tauri/src/lib.rs (hunk 3 above)
Comment (Alystrasz) [reply src-tauri/src/lib.rs:485]: If it works, it works™️

**Pull request merged**